test_crates/enum_struct_union_deprecated_added/new/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+// STRUCTS
```

Comment:
Absolutely love the thorough test cases here!

For easier readability and maintainability, consider splitting up the contents here across three files: `structs.rs, enums.rs, unions.rs`. Then `lib.rs` can just do: 
```rust
pub mod structs;
pub mod enums;
pub mod unions;
```
And everything should work correctly.

This will make the spans a bit easier to work with, since we won't have to scroll deep into the file to find the reported line, and it's less likely that a test fails just because an item moved up or down a line and changed the reported span.